src/dsp_tools/models/xmlvalue.py

```diff
@@ -23,14 +24,82 @@ def __init__(
         self.comment = node.get("comment")
         self.permissions = node.get("permissions")
         if val_type == "text" and node.get("encoding") == "xml":
-            node.attrib.clear()
-            xmlstr = etree.tostring(node, encoding="unicode", method="xml")
-            xmlstr = xmlstr.replace('<text xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">', "")
-            xmlstr = xmlstr.replace("</text>", "")
-            self.value = KnoraStandoffXml(xmlstr)
+            xmlstr_orig = etree.tostring(node, encoding="unicode", method="xml")
+            xmlstr_cleaned = self._cleanup_formatted_text(xmlstr_orig)
+            self.value = KnoraStandoffXml(xmlstr_cleaned)
             self.resrefs = list({x.split(":")[1] for x in self.value.get_all_iris() or []})
+        elif val_type == "text" and node.get("encoding") == "utf8":
+            str_orig = "".join(node.itertext())
+            str_cleaned = self._cleanup_unformatted_text(str_orig)
+            self.value = str_cleaned
         elif val_type == "list":
             listname = cast(str, listname)
             self.value = listname + ":" + "".join(node.itertext())
         else:
             self.value = "".join(node.itertext())
+
+    def _cleanup_formatted_text(self, xmlstr_orig: str) -> str:
+        """
+        In a xml-encoded text value from the XML file,
+        there may be non-text characters that must be removed.
+        This method:
+            - removes the <text> tags
+            - replaces (multiple) line breaks by a space
+            - replaces multiple spaces or tabstops by a single space (except within <code> tags)
+
+        Args:
+            xmlstr_orig: original string from the XML file
+
+        Returns:
+            purged string, suitable to be sent to DSP-API
+        """
+        # remove the <text> tags
+        xmlstr = regex.sub("<text.*?>", "", xmlstr_orig)
+        xmlstr = regex.sub("</text>", "", xmlstr)
+
+        # replace (multiple) line breaks by a space
+        xmlstr = regex.sub("\n+", " ", xmlstr)
+
+        # replace multiple spaces or tabstops by a single space (except within <code> tags)
+        # the regex selects all spaces/tabstops not followed by </code> without <code in between.
+        # credits: https://stackoverflow.com/a/46937770/14414188
+        xmlstr = regex.sub("( {2,}|\t+)(?!(.(?!<code))*</code>)", " ", xmlstr)
+
+        # remove spaces after <br/> tags (except within <code> tags)
+        xmlstr = regex.sub("((?<=<br/?>) )(?!(.(?!<code))*</code>)", "", xmlstr)
+
+        # remove leading and trailing spaces
+        xmlstr = xmlstr.strip()
+
+        return xmlstr
```

Comment:
I feel the same than Nora: It would perhaps be possible to do this with lxml, but I would be super scared to miss edge cases. I will never now if the `etree.Element` has children or not, and I think that the logic would become super complicated if I try to do the same with lxml. Perhaps it's just that I have more confidence in my regex skills than in my lxml skills 😄 .

Reply:
fair enough :)